relay-server/src/actors/project_cache.rs

```diff
@@ -85,8 +86,18 @@ impl ProjectCache {
         let eviction_start = Instant::now();
         let delta = 2 * self.config.project_cache_expiry() + self.config.project_grace_period();
 
-        self.projects
-            .retain(|_, entry| entry.last_updated_at() + delta > eviction_start);
+        let expired = self
+            .projects
+            .drain_filter(|_, entry| entry.last_updated_at() + delta <= eviction_start);
+
+        // Defer dropping the projects to a dedicated thread:
+        for (_, project) in expired {
+            self.garbage_disposal.dispose(project);
+        }
+
+        // Log garbage queue size:
+        let queue_size = self.garbage_disposal.queue_size() as f64;
```

Comment:
this alone was enough reason for me to report queue size as u64 rather than usize :)

Reply:
Can you clarify? You could equally have the queue_size as `usize` here and then cast this to an f64.

Reply:
Wait, why is this `f64` and not `u64`?  Only just noticed that and now I'm confused.

- Metrics are reported as `u64`.
- We don't really know what sizes a channel can get up to as this is not described AFAIK, but we can probably assume we'd crash before we reach the max size.
- That means deciding to expose it as a `usize` is arbitrary as exposing it as a `u64`.
- But now here you are lead to believe that the true value is represented by a `usize` and you're supposedly casting here.  This is misleading.